app/frontend/src/api/api.ts

```diff
@@ -23,9 +23,10 @@ export async function configApi(): Promise<Config> {
 }
 
 export async function askApi(request: ChatAppRequest, idToken: string | undefined): Promise<ChatAppResponse> {
+    const headers = await getHeaders(idToken);
     const response = await fetch(`${BACKEND_URI}/ask`, {
         method: "POST",
-        headers: { ...getHeaders(idToken), "Content-Type": "application/json" },
+        headers: { ...headers, "Content-Type": "application/json" },
```

Comment:
Was this split into 2 lines for debuggability, or because we can't use await with the ...? Just curious.

Reply:
I assumed we couldn't use the ... with await, but I can put it back to ...(await getHeaders(idToken)) if necessary

Reply:
I don't mind either way, I think splitting for debuggability can be nice anyway.

Reply:
OK - I'll leave this as-is